src/Domain/DonationService.php

```diff
@@ -25,7 +25,7 @@
 
 readonly class DonationService
 {
-    private const MAX_RETRY_COUNT = 4;
+    private const MAX_RETRY_COUNT = 3;
```

Comment:
Given some failure modes have slightly increased negative impacts by retrying a lot of times, and I can't remember seeing anything in donation setup succeeding after a 3rd time or later, I think 3 is a good number for now. Not set on the change though!